newm-server/src/main/kotlin/io/newm/server/features/distribution/eveara/EvearaDistributionRepositoryImpl.kt

```diff
@@ -1445,6 +1445,13 @@ class EvearaDistributionRepositoryImpl(
 //        val simulateReleaseResponse = simulateDistributeRelease(user, mutableSong.distributionReleaseId!!)
 //        log.info { "Simulated release ${mutableSong.title} with id ${mutableSong.distributionReleaseId}: ${simulateReleaseResponse.message}" }
 
+        // Verify we have enough days to distribute the release (in case this is a re-process and some days have passed)
+        val earliestReleaseDate = getEarliestReleaseDate(user.id)
+        if (earliestReleaseDate.isAfter(mutableSong.releaseDate!!)) {
+            mutableSong = mutableSong.copy(releaseDate = earliestReleaseDate)
+            songRepository.update(mutableSong.id!!, mutableSong)
```

Comment:
I'm looking at this `update()` function. It seems it's never possible to set a field back to null once it's been set. I think there are a few potential bugs related to this. I see one area I set a value to null, but that means it's not actually getting set.

Reply:
The `update()` needs to behave lol je that because it's used for the PATCH Api. Maybe we can create a separate `set()` function that will set all values and use that one where needed.

Reply:
Please look at the new `editFields` method I created.